internal/controller/datadogagent/controller_v2_test.go

```diff
@@ -1100,6 +1100,248 @@ func Test_AutopilotOverrides(t *testing.T) {
+			},
+		},
+		{
+			// This test verifies that when a node has a GKE provider label with an unsupported OS value,
+			// the system falls back to the "default" provider for control plane monitoring
+			name: "[introspection] Control Plane Monitoring with unsupported provider",
+			fields: fields{
+				scheme:   s,
+				recorder: recorder,
+			},
+			loadFunc: func(c client.Client) *v2alpha1.DatadogAgent {
+				return createDatadogAgentWithClusterChecks(c, resourcesNamespace, resourcesName)
+			},
+			nodes: []client.Object{
+				&corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "gke-node-1",
+						Labels: map[string]string{
+							// Use unsupported OS value to trigger fallback to "default" provider
+							kubernetes.GKEProviderLabel: "unsupported-os",
+						},
+					},
+				},
+			},
+			want:    reconcile.Result{RequeueAfter: defaultRequeueDuration},
+			wantErr: false,
+			wantFunc: func(t *testing.T, c client.Client) error {
+				return verifyDCADeployment(t, c, resourcesName, resourcesNamespace, dcaName, "default")
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &Reconciler{
+				client:       fake.NewClientBuilder().WithStatusSubresource(&corev1.Node{}, &v2alpha1.DatadogAgent{}).WithObjects(tt.nodes...).Build(),
+				scheme:       tt.fields.scheme,
+				platformInfo: tt.fields.platformInfo,
+				recorder:     recorder,
+				log:          logf.Log.WithName(tt.name),
+				forwarders:   forwarders,
+				options: ReconcilerOptions{
+					ExtendedDaemonsetOptions: componentagent.ExtendedDaemonsetOptions{
+						Enabled: false,
+					},
+					SupportCilium:        false,
+					IntrospectionEnabled: true,
+				},
+			}
+
+			var dda *v2alpha1.DatadogAgent
+			if tt.loadFunc != nil {
+				dda = tt.loadFunc(r.client)
+			}
+			got, err := r.Reconcile(context.TODO(), dda)
+			if tt.wantErr {
+				assert.Error(t, err, "ReconcileDatadogAgent.Reconcile() expected an error")
+			} else {
+				assert.NoError(t, err, "ReconcileDatadogAgent.Reconcile() unexpected error: %v", err)
+			}
+
+			assert.Equal(t, tt.want, got, "ReconcileDatadogAgent.Reconcile() unexpected result")
+
+			if tt.wantFunc != nil {
+				err := tt.wantFunc(t, r.client)
+				assert.NoError(t, err, "ReconcileDatadogAgent.Reconcile() wantFunc validation error: %v", err)
+			}
+		})
+	}
+}
+
+func verifyDCADeployment(t *testing.T, c client.Client, ddaName, resourcesNamespace, expectedName string, provider string) error {
+	daemonSetList := appsv1.DeploymentList{}
+	if err := c.List(context.TODO(), &daemonSetList, client.HasLabels{constants.MD5AgentDeploymentProviderLabelKey}); err != nil {
+		return err
+	}
+	assert.Equal(t, 1, len(daemonSetList.Items))
+	assert.Equal(t, expectedName, daemonSetList.Items[0].ObjectMeta.Name)
+
+	cms := corev1.ConfigMapList{}
+	if err := c.List(context.TODO(), &cms, client.InNamespace(resourcesNamespace)); err != nil {
+		return err
+	}
+	cpCm := corev1.ConfigMap{}
```

Comment:
should we assert anything on cpCm?

Reply:
We could but I think that would be redundant testing, since we already verify in `internal/controller/datadogagent/feature/controlplanemonitoring/configmap_test.go` that the contents of the configmap is accurate corresponding to the provider. Here we just assert that the correct configmap is created.